crowbar_framework/app/models/service_object.rb

```diff
@@ -1168,6 +1168,9 @@ def apply_role(role, inst, in_queue)
     # of names of Chef roles.
     run_order = []
 
+    # get databag to remember potential removal of a role
+    databag = ProposalObject.find_proposal(@bc_name, inst)
```

Comment:
Shouldn't we read from the databag to also populate `old_nodes`? I mean, the goal for this hack is to make sure chef-client is run for the removed nodes in case there's a crash. And based on the code here, I don't see that happening -- or am I missing something?